kpi/deployment_backends/base_backend.py

```diff
@@ -224,3 +225,21 @@ def get_submission(self, pk, requesting_user_id,
         except IndexError:
             pass
         return None
+
+    def set_status(self, status):
+        # Avoid circular imports
+        # use `self.asset.__class__` instead of `from kpi.models import Asset`
+        self.asset.__class__.objects.filter(id=self.asset.pk).update(
+            _deployment_data=ReplaceValue(
```

Comment:
discussion notes—the problem this is trying to solve is:
1. uWSGI process running `kpi.serializers.v1.deployment.create()` has an in-memory copy of asset a123
2. `kpi.serializers.v1.deployment.create()` calls `sync_media_files.delay()` for a123
3. celery completes `sync_media_files` quickly, calling `asset.deployment.set_status(asset.deployment.STATUS_SYNCED)`
4. `kpi.serializers.v1.deployment.create()` continues, eventually overwriting the entirety of a123's `_deployment_data`, removing the `status` attribute

Reply:
i'll argue that we should still use row-level locking because otherwise we have to rely on everyone always remembering to use `ReplaceValue` each time they touch `_deployment_data`. example of another place:
https://github.com/kobotoolbox/kpi/blob/f3985ae3f6d1c9d264ce3c618ea062f4a5f02f18/kpi/serializers/v1/deployment.py#L73-L80

actually, you anticipated this already:
https://github.com/kobotoolbox/kpi/blob/f3985ae3f6d1c9d264ce3c618ea062f4a5f02f18/kpi/models/asset.py#L975-L987

this reduces the window of opportunity for the race condition, but we could still have a problem if celery changes `_deployment_data` while another process is between lines 983 and 987 above

Reply:
another question is whether or not it's ever safe to call `Asset.save(refresh_status=False)`

Reply:
All of this has been refactored in https://github.com/kobotoolbox/kpi/pull/2537/commits/314679415003a70decbb5e9771a42e952d1ab157